### PR TITLE
feat: api languages postgres

### PR DESCRIPTION
**Pull request by mikeallisonJS**
# Description

### <samp>🤖 Generated by Copilot at ef7924e</samp>

This pull request migrates the `api-languages` app from ArangoDB to PostgreSQL using Prisma as the database layer. It replaces the `DatabaseModule`, the `CountryService`, and the `LanguageService` with the `PrismaService`, and refactors the `CountryResolver` and the `LanguageResolver` to use the Prisma client methods. It also updates the `seed.ts` file, the `project.json` file, the test files, the Dockerfile, the Terraform configuration, and the environment variables to work with Prisma and PostgreSQL. It adds new files for the Prisma schema, migrations, and client generation. It deletes the `db.ts` file and the service files that are no longer needed.

- Link to Basecamp Todo

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [ ] No noticeable change
- [ ] Seed functions as expected

# Walkthrough

### <samp>🤖 Generated by Copilot at ef7924e</samp>

*  Switch the database from ArangoDB to PostgreSQL using Prisma for the `api-languages` app ([link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-5d50bd2bee8b823b3f8b5ce6ea83e272b32e7d083df58cbd2ff715f82edb7dbbR1-R2), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-0f8cb5f3f2891033b92ba2a7b0e3c4a709b52202522bc8914eac99a836316e03L2-R15), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-5cb6504b4b21938ef03c1e27b38f96a2f221cd492c96aad43348a31495876396R1-R49), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-c517f5c6688479a1296a784fbb40e07218209f75929cadb6dcdd9dbc7c088379L1-R2), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-e00645ae9686203d2dc5eb9a40d0a82f15caa7297d886e8b5416baa0efb8858aR1-R34), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-0e0145704627be65e90fd5d2519f9bc915688ec163f08b812855a394be43cf96L4-R14), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-0e0145704627be65e90fd5d2519f9bc915688ec163f08b812855a394be43cf96L29-R37), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-0e0145704627be65e90fd5d2519f9bc915688ec163f08b812855a394be43cf96L67-R78), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-0e0145704627be65e90fd5d2519f9bc915688ec163f08b812855a394be43cf96L106-R98), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-0e0145704627be65e90fd5d2519f9bc915688ec163f08b812855a394be43cf96L117-R114), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-0e0145704627be65e90fd5d2519f9bc915688ec163f08b812855a394be43cf96L140-R146), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-0e0145704627be65e90fd5d2519f9bc915688ec163f08b812855a394be43cf96L195-R186), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-0e0145704627be65e90fd5d2519f9bc915688ec163f08b812855a394be43cf96L201-R192), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-0e0145704627be65e90fd5d2519f9bc915688ec163f08b812855a394be43cf96L224-R209), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-0e0145704627be65e90fd5d2519f9bc915688ec163f08b812855a394be43cf96L240-R225), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-0e0145704627be65e90fd5d2519f9bc915688ec163f08b812855a394be43cf96L255-R232), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-91fd6f362c107708e04b5971ac24631ac9463627eafa4039cea4dad6646c8ba5R8), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-31184e2eaf6f670fdaf995f1c7c0e9252a700c25768c93831aa0e9f746422ee2L19-R34), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-11d2a7cddf4dfb854ffc0395d89cf7641f4b993ff5943e6c4efbe17f8851da94R40-R46), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-e3e54f41d848586c822dcaccbcce0a5f3ac494fa126d831daf566b7975d73c9cL37-L42), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-e3e54f41d848586c822dcaccbcce0a5f3ac494fa126d831daf566b7975d73c9cR82-R134), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-9607d888ad8233f0dbbf4a64b08abedc4748e07629eb871a5539ad9c38a7e8daR1-R15), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-647cca9c0c32b04fa4e0f172ea8b4faa47a8b37fe69dc0b67d047522f992c3ceL2-R8), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-dc0f50d38101e3b6d7f77697090bee2388211c2509e467a21e06e22dd0e342e8L10-R21), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-dc0f50d38101e3b6d7f77697090bee2388211c2509e467a21e06e22dd0e342e8L30-R31), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-dc0f50d38101e3b6d7f77697090bee2388211c2509e467a21e06e22dd0e342e8L63-R64), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-dc0f50d38101e3b6d7f77697090bee2388211c2509e467a21e06e22dd0e342e8L71-R74), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-69e677dee1a0487bf2d34cfd9170c93664caeea92477ccc81626f355a1e03fe6L2-R8), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-8f68509343007c43b016432a4c618e12243edc99df9a62e47821edd9c719fa5cL2-R6), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-8f68509343007c43b016432a4c618e12243edc99df9a62e47821edd9c719fa5cL22-R36), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-8f68509343007c43b016432a4c618e12243edc99df9a62e47821edd9c719fa5cL43-R45), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-45f30fb7c4deec8e8440b7cc6d6d613470485dd6de5534f68babd2e80de558b8L10-R17), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-45f30fb7c4deec8e8440b7cc6d6d613470485dd6de5534f68babd2e80de558b8L21-R27), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-45f30fb7c4deec8e8440b7cc6d6d613470485dd6de5534f68babd2e80de558b8L29-R37), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-45f30fb7c4deec8e8440b7cc6d6d613470485dd6de5534f68babd2e80de558b8L47-R55), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-28747e0e6f0f037fd5d4646cb5f719c63eb45de1f660281786d0b0411e8ed0b0), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-1f903e05991349d74f8777f9421f8f4cc3b3ff30b73e8f90346128833c75d07a), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-95bc736792982a589ec0d6733fb7a97e47e0d60263036ba12834e4ba0b749c0e))
  * Add environment variables for the PostgreSQL database URL and the Prisma client output location to the `.prisma-generate.env` file ([link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-5d50bd2bee8b823b3f8b5ce6ea83e272b32e7d083df58cbd2ff715f82edb7dbbR1-R2))
  * Modify the `Dockerfile` to install the Prisma CLI and the Prisma client, run the Prisma generate and migrate commands, and add the `openssl1.1-compat` package ([link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-0f8cb5f3f2891033b92ba2a7b0e3c4a709b52202522bc8914eac99a836316e03L2-R15))
  * Add a new file `migration.sql` to the `db/migrations` folder, which contains the SQL commands to create the tables and indexes for the `Country` and `Language` models ([link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-5cb6504b4b21938ef03c1e27b38f96a2f221cd492c96aad43348a31495876396R1-R49))
  * Add a new file `migration_lock.toml` to the `db/migrations` folder, which contains the metadata for the Prisma migrations ([link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-c517f5c6688479a1296a784fbb40e07218209f75929cadb6dcdd9dbc7c088379L1-R2))
  * Add a new file `schema.prisma` to the `db` folder, which contains the Prisma schema for the `Country` and `Language` models ([link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-e00645ae9686203d2dc5eb9a40d0a82f15caa7297d886e8b5416baa0efb8858aR1-R34))
  * Modify the `seed.ts` file to import the Prisma client and types, use the Prisma client methods to query and update the database, and align the interfaces with the Prisma models ([link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-0e0145704627be65e90fd5d2519f9bc915688ec163f08b812855a394be43cf96L4-R14), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-0e0145704627be65e90fd5d2519f9bc915688ec163f08b812855a394be43cf96L29-R37), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-0e0145704627be65e90fd5d2519f9bc915688ec163f08b812855a394be43cf96L67-R78), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-0e0145704627be65e90fd5d2519f9bc915688ec163f08b812855a394be43cf96L106-R98), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-0e0145704627be65e90fd5d2519f9bc915688ec163f08b812855a394be43cf96L117-R114), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-0e0145704627be65e90fd5d2519f9bc915688ec163f08b812855a394be43cf96L140-R146), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-0e0145704627be65e90fd5d2519f9bc915688ec163f08b812855a394be43cf96L195-R186), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-0e0145704627be65e90fd5d2519f9bc915688ec163f08b812855a394be43cf96L201-R192), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-0e0145704627be65e90fd5d2519f9bc915688ec163f08b812855a394be43cf96L224-R209), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-0e0145704627be65e90fd5d2519f9bc915688ec163f08b812855a394be43cf96L240-R225), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-0e0145704627be65e90fd5d2519f9bc915688ec163f08b812855a394be43cf96L255-R232))
  * Add the `PG_DATABASE_URL` environment variable to the `locals.tf` file in the `infrastructure` folder ([link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-91fd6f362c107708e04b5971ac24631ac9463627eafa4039cea4dad6646c8ba5R8))
  * Modify the `main.tf` file in the `infrastructure` folder to add the `PG_DATABASE_URL` environment variable to the `ecs_service` module and to add the `database` module, which creates the Aurora PostgreSQL database ([link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-31184e2eaf6f670fdaf995f1c7c0e9252a700c25768c93831aa0e9f746422ee2L19-R34))
  * Add the `subnet_group_name` and the `vpc_security_group_id` variables to the `variables.tf` file in the `infrastructure` folder, which are used by the `database` module ([link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-11d2a7cddf4dfb854ffc0395d89cf7641f4b993ff5943e6c4efbe17f8851da94R40-R46))
  * Remove the `seed` target from the `project.json` file and add several new targets to run the Prisma CLI commands ([link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-e3e54f41d848586c822dcaccbcce0a5f3ac494fa126d831daf566b7975d73c9cL37-L42), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-e3e54f41d848586c822dcaccbcce0a5f3ac494fa126d831daf566b7975d73c9cR82-R134))
  * Add a new file `prisma.service.ts` to the `app/lib` folder, which contains the code for the `PrismaService` class that extends the Prisma client and connects to the database ([link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-9607d888ad8233f0dbbf4a64b08abedc4748e07629eb871a5539ad9c38a7e8daR1-R15))
  * Modify the `country.module.ts` file to remove the `DatabaseModule` and the `LanguageService` and to add the `PrismaService` ([link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-647cca9c0c32b04fa4e0f172ea8b4faa47a8b37fe69dc0b67d047522f992c3ceL2-R8))
  * Modify the `country.resolver.ts` file to remove the `CountryService` and the `LanguageService` and to use the Prisma client methods to query the database ([link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-dc0f50d38101e3b6d7f77697090bee2388211c2509e467a21e06e22dd0e342e8L10-R21), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-dc0f50d38101e3b6d7f77697090bee2388211c2509e467a21e06e22dd0e342e8L30-R31), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-dc0f50d38101e3b6d7f77697090bee2388211c2509e467a21e06e22dd0e342e8L63-R64), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-dc0f50d38101e3b6d7f77697090bee2388211c2509e467a21e06e22dd0e342e8L71-R74))
  * Modify the `language.module.ts` file to remove the `DatabaseModule` and the `LanguageService` and to add the `PrismaService` ([link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-69e677dee1a0487bf2d34cfd9170c93664caeea92477ccc81626f355a1e03fe6L2-R8))
  * Modify the `language.resolver.ts` file to remove the `LanguageService` and to use the Prisma client methods to query the database ([link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-45f30fb7c4deec8e8440b7cc6d6d613470485dd6de5534f68babd2e80de558b8L10-R17), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-45f30fb7c4deec8e8440b7cc6d6d613470485dd6de5534f68babd2e80de558b8L21-R27), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-45f30fb7c4deec8e8440b7cc6d6d613470485dd6de5534f68babd2e80de558b8L29-R37), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-45f30fb7c4deec8e8440b7cc6d6d613470485dd6de5534f68babd2e80de558b8L47-R55))
  * Delete the `db.ts` file, which contained the code for the ArangoDB connection and configuration ([link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-28747e0e6f0f037fd5d4646cb5f719c63eb45de1f660281786d0b0411e8ed0b0))
  * Delete the `country.service.spec.ts` and `country.service.ts` files, which contained the code and tests for the `CountryService` class that used the ArangoDB client ([link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-1f903e05991349d74f8777f9421f8f4cc3b3ff30b73e8f90346128833c75d07a), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-95bc736792982a589ec0d6733fb7a97e47e0d60263036ba12834e4ba0b749c0e))
* Modify the tests for the `CountryResolver` and the `LanguageResolver` classes to use the `PrismaService` mock and to expect the Prisma client methods to be called ([link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-b5a8f8ee6bdcdc8d36322319a97ca36f5d0c6dd75b825639738088ac073bb7d1L2-R7), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-b5a8f8ee6bdcdc8d36322319a97ca36f5d0c6dd75b825639738088ac073bb7d1L24-R29), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-b5a8f8ee6bdcdc8d36322319a97ca36f5d0c6dd75b825639738088ac073bb7d1L47-R35), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-b5a8f8ee6bdcdc8d36322319a97ca36f5d0c6dd75b825639738088ac073bb7d1L54-R51), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-8f68509343007c43b016432a4c618e12243edc99df9a62e47821edd9c719fa5cL22-R36), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-8f68509343007c43b016432a4c618e12243edc99df9a62e47821edd9c719fa5cL43-R45))
  * Modify the `country.resolver.spec.ts` file to remove the `CountryService`, the `LanguageService`, and the `Database` mock and to add the `PrismaService` mock, and to expect the `prismaService.country.findUnique` and `prismaService.country.findMany` methods to be called ([link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-b5a8f8ee6bdcdc8d36322319a97ca36f5d0c6dd75b825639738088ac073bb7d1L2-R7), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-b5a8f8ee6bdcdc8d36322319a97ca36f5d0c6dd75b825639738088ac073bb7d1L24-R29), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-b5a8f8ee6bdcdc8d36322319a97ca36f5d0c6dd75b825639738088ac073bb7d1L47-R35), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-b5a8f8ee6bdcdc8d36322319a97ca36f5d0c6dd75b825639738088ac073bb7d1L54-R51))
  * Modify the `language.resolver.spec.ts` file to remove the `LanguageService` and to add the `PrismaService` mock, and to expect the `prismaService.language.findUnique` and `prismaService.language.findMany` methods to be called ([link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-8f68509343007c43b016432a4c618e12243edc99df9a62e47821edd9c719fa5cL22-R36), [link](https://github.com/JesusFilm/core/pull/1678/files?diff=unified&w=0#diff-8f68509343007c43b016432a4c618e12243edc99df9a62e47821edd9c719fa5cL43-R45))

### CODE REVIEW

File: apps/api-languages/db/schema.prisma [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+model Language {
+  id        String    @id
+  createdAt DateTime  @default(now())
+  updatedAt DateTime  @updatedAt
+  name      Json[]
+  bcp47     String?
+  iso3      String?
+  Countries Country[]
```
Comment (tataihono) [inline apps/api-languages/db/schema.prisma:33]:
```suggestion
  countries Country[]
```

File: apps/api-languages/db/schema.prisma [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  createdAt   DateTime   @default(now())
+  updatedAt   DateTime   @updatedAt
+  name        Json[]
+  population  Int
+  slug        String     @unique
+  latitude    Float
+  longitude   Float
+  image       String
+  continent   Json[]
+  languageIds String[]
```
Comment (tataihono) [inline apps/api-languages/db/schema.prisma:22]: this shouldn't be stored in an array

**Review by tataihono — CHANGES_REQUESTED** [review]

File: apps/api-languages/db/schema.prisma [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  id          String     @id
+  createdAt   DateTime   @default(now())
+  updatedAt   DateTime   @updatedAt
+  name        Json[]
+  population  Int
+  slug        String     @unique
+  latitude    Float
+  longitude   Float
+  image       String
+  continent   Json[]
```
Comment (tataihono) [inline apps/api-languages/db/schema.prisma:21]: can we move this into a separate model?

**Review by tataihono — CHANGES_REQUESTED** [review]

**Comment by mikeallisonJS** [comment]
atlantis unlock

**Review by tataihono — APPROVED** [review]

**Review by tataihono — APPROVED** [review]

**Pull request merged**